api/api_public_blockchain.go

```diff
@@ -164,10 +164,17 @@ func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.H
 // transactions in the block are returned in full detail, otherwise only the transaction hash is returned.
 func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, blockNr rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
 	block, err := s.b.BlockByNumber(ctx, blockNr)
-	if err != nil {
-		return nil, err
+	if block != nil && err == nil {
```

Comment:
@hqjang-pepper It seems when block and err are both nil, it returns nil, nil. Also, is it possible that block is nil?

Reply:
Since there is no case of return nil,nil in BlockByNumber at api_backend.go.
So return nil, nil is needless.

Reply:
When block is nil, it always return err

Reply:
@hqjang-pepper According to https://github.com/klaytn/klaytn/pull/738/files, is it okay not to change `GetBlockByHash`?

Reply:
@jeongkyun-oh Revision of `GetBlockByHash` is not related to pending block issue, so I think it's unnecessary.